### DISCUSSION TITLE
How to use arguments with spaces and dots when using $os.cmd() to execute shell commands in pb_hooks?

### DISCUSSION THREAD

**Discussion by AfdhalTan** [pocketbase/pocketbase#4945] [Q&A] [answered]
Hello,

I am trying to get an output from this:
```bash
nmcli --terse --color no --mode multiline --show-secrets --wait 10 --fields ipv4.method connection show "Wired connection 1"
```

Using hook:
```js
const arg_nmcli = ['--terse', '--color', 'no', '--mode', 'multiline', '--show-secrets', '--wait', '10'];
const connection = 'Wired connection 1';
const result = String.fromCharCode(...$os.cmd('nmcli', ...arg_nmcli, ...['--fields', 'ipv4.method', 'connection', 'show', `"${connection}"`]).output());
console.log(result);
```

I got an error.

Any suggestions?

**Comment by ganigeorgiev** [comment]
> I got an error.

What is the error?
Are you sure that the PocketBase user that started the process is able to run nmcli?

In general the above double array spread/expansion looks weird but I guess it should work.

**Reply by AfdhalTan** [reply]
The error in the log is:
```
exit status 10
```

Using this to print:
```js
console.log($os.cmd('nmcli', ...).string());
```

In Pocketbase's hook this works fine:
```bash
nmcli --terse --color no --mode multiline --show-secrets --wait 10 device show
```

This doesn't:
```bash
nmcli --terse --color no --mode multiline --show-secrets --wait 10 --fields ipv4.method connection show "Wired connection 1"
```
The difference that I can see is `ipv4.method` has a dot and `"Wired connection 1"` has some spaces.

Both commands works fine in shell.

**Reply by ganigeorgiev** [reply]
For me the last command also returns `10` as exit code when running directly from the terminal (_most likely because I don't have such connection_).

I'm not sure what could be the issue in your case but `$os.cmd` is a direct mapping to the Go's [`os/exec.Command`](https://pkg.go.dev/os/exec#Command).

**Reply by AfdhalTan** [reply]
Thank you for the fast answer and the link.

I will investigate `os/exec.Command`.

**Reply by ganigeorgiev** [reply, marked as answer]
I also just noticed that you are inserting extra double quotes around the "Wired connection 1" string. Try to remove them as these quotes are usually needed only for the shell to properly split/extract the args.

**Reply by AfdhalTan** [reply]
You are right, silly me 😅

Thank you
